src/stx/stl/trex_stl_dp_core.cpp

```diff
@@ -807,16 +807,28 @@ bool PerPortProfile::stop_traffic(uint8_t  port_id,
         }
     }
 
+
+    // TODO: refactor me (Mateusz, 2019-11-07)
     for (auto& dp_stream : m_active_nodes) {
         CGenNodeStateless * node =dp_stream.m_node;
         assert(node->get_port_id() == port_id);
-        if ( node->get_state() == CGenNodeStateless::ss_ACTIVE) {
-            node->mark_for_free();
-            m_active_streams--;
-            dp_stream.DeleteOnlyStream();
-
-        }else{
-            dp_stream.Delete(m_port->m_core);
+        if (node->m_type != CGenNode::TIMESYNC) {
```

Comment:
Why? it seems it is the same code why CGenNodeTimesync needs a diff treatment

Reply:
Sure, I am working on a refactor.